client/hotrod-client/src/test/java/org/infinispan/client/hotrod/query/IndexedCacheNonIndexedEntityTest.java

```diff
@@ -0,0 +1,112 @@
+package org.infinispan.client.hotrod.query;
+
+import static org.infinispan.client.hotrod.test.HotRodClientTestingUtil.killRemoteCacheManager;
+import static org.infinispan.client.hotrod.test.HotRodClientTestingUtil.killServers;
+import static org.infinispan.client.hotrod.test.HotRodClientTestingUtil.newRemoteConfigurationBuilder;
+import static org.infinispan.commons.api.CacheContainerAdmin.AdminFlag.VOLATILE;
+import static org.infinispan.query.remote.client.ProtobufMetadataManagerConstants.PROTOBUF_METADATA_CACHE_NAME;
+import static org.testng.AssertJUnit.assertFalse;
+
+import java.io.IOException;
+
+import org.infinispan.client.hotrod.RemoteCache;
+import org.infinispan.client.hotrod.RemoteCacheManager;
+import org.infinispan.client.hotrod.exceptions.HotRodClientException;
+import org.infinispan.client.hotrod.marshall.MarshallerUtil;
+import org.infinispan.client.hotrod.test.HotRodClientTestingUtil;
+import org.infinispan.commons.configuration.XMLStringConfiguration;
+import org.infinispan.configuration.cache.ConfigurationBuilder;
+import org.infinispan.manager.EmbeddedCacheManager;
+import org.infinispan.protostream.SerializationContext;
+import org.infinispan.protostream.annotations.ProtoFactory;
+import org.infinispan.protostream.annotations.ProtoField;
+import org.infinispan.protostream.annotations.ProtoName;
+import org.infinispan.protostream.annotations.ProtoSchemaBuilder;
+import org.infinispan.query.remote.client.ProtobufMetadataManagerConstants;
+import org.infinispan.server.core.admin.embeddedserver.EmbeddedServerAdminOperationHandler;
+import org.infinispan.server.hotrod.HotRodServer;
+import org.infinispan.server.hotrod.configuration.HotRodServerConfigurationBuilder;
+import org.infinispan.test.SingleCacheManagerTest;
+import org.infinispan.test.fwk.TestCacheManagerFactory;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.Test;
+
+/**
+ * @since 12.0
+ */
+@Test(testName = "client.hotrod.query.IndexedCacheNonIndexedEntityTest", groups = "functional")
+public class IndexedCacheNonIndexedEntityTest extends SingleCacheManagerTest {
+   private static final String CACHE_NAME = "IndexedCacheNonIndexedEntitiesTest";
```

Comment:
can we consider using the new class rules for this test ?

Reply:
What are "new class rules"?

Reply:
The Junit 4 Rules used for testing

Reply:
I don't know if we can use them or not in this test